File: 剑指offer/32.2.从上到下打印二叉树.py
from typing import List


class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def levelOrder(self, root: TreeNode) -> List[int]:
        ans = []

        if not root:
            return []

        queue = [root]
        while queue:
            tmp = []
            for _ in range(len(queue)):
                cur = queue.pop(0)
                tmp.append(cur.val)

                if cur.left:
                    queue.append(cur.left)
                if cur.right:
                    queue.append(cur.right)
            ans.append(tmp)

        return ans

File: 剑指offer/32.2.从上到下打印二叉树.cpp
#include <deque>
#include <iostream>
#include <vector>

using namespace std;

struct TreeNode {
    int val;
    TreeNode* left;
    TreeNode* right;
    TreeNode(int x)
        : val(x)
        , left(NULL)
        , right(NULL)
    {
    }
};

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root)
    {
        vector<vector<int>> ans;

        if (root == nullptr) {
            return ans;
        }

        deque<TreeNode*> queue;

        queue.push_back(root);

        while (not queue.empty()) {
            vector<int> tmp;

            int n = queue.size();
            for (int i = 0; i < n; i++) {
                TreeNode* cur = queue.front();
                queue.pop_front();

                tmp.push_back(cur->val);

                if (cur->left) {
                    queue.push_back(cur->left);
                }

                if (cur->right) {
                    queue.push_back(cur->right);
                }
            }

            ans.push_back(tmp);
        }

        return ans;
    }
};